base_layer/core/src/mempool/service/inbound_handlers.rs

```diff
@@ -62,6 +62,9 @@ where T: BlockchainBackend + 'static
             MempoolRequest::GetStats => Ok(MempoolResponse::Stats(
                 async_mempool::stats(self.mempool.clone()).await?,
             )),
+            MempoolRequest::GetState => Ok(MempoolResponse::State(
+                async_mempool::state(self.mempool.clone()).await?,
```

Comment:
A little wary of this -- The mempool state could become huge, and requesting it over the wire could be a simple DOS vector

Reply:
That is true, we can remove it, or change it, in the near future